go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy.go

```diff
@@ -83,7 +83,7 @@ func (c *Controller) syncAdminNetworkPolicy(key string) error {
 	err = c.ensureAdminNetworkPolicy(anp)
 	if err != nil {
 		// we can ignore the error if status update doesn't succeed; best effort
-		_ = c.updateANPStatusToNotReady(anp, c.zone, err.Error())
+		_ = c.updateANPStatusToNotReady(anp.Name, err.Error())
```

Comment:
Overall it looks good for server side apply.
one question: this is having condition per zone, doesn't it also need to have another condition object in cluster manager which can convey overall status of the object ?

Reply:
yeah great question Peri! the problem here is that unlike the other feature CRDs (EQoS, EFW, APBR) which we own in ovnkube with which we can have custom statuses, with ANP we do upstream first: https://github.com/kubernetes-sigs/network-policy-api/blob/8652dfdbb7e5563a2800716996417e4a4ceaf399/apis/v1alpha1/adminnetworkpolicy_types.go#L49 so the CRD lives there which is used by other plugins outside of OVNK, so we cannot add OVNK specific API fields for status, a bit restricted there BUT, I think this ok for now, we did do a scale test (without server side apply) on 100 nodes which looked like:

```
$ oc describe anp ama | grep horsley
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-tf2ww
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-ttxmc
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-lndzj
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-h5thc
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-qhrwc
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-master-0
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-pkmxk
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-5lhtz
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-gq9wt
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-z2h5d
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-x65wb
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-6wtxt
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-b-xxcrt
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-4fkjd
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-master-2
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-5wz8f
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-nb876
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-74nn5
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-bld4w
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-rv44f
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-f85n5
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-twlhv
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-rq7gj
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-7whb2
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-xjp8q
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-jntn4
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-wh22z
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-rwbdh
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-bdj65
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-qfqkv
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-8wb77
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-qftlq
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-vfdbm
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-sxnvs
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-zxlg5
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-mkrkw
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-6v6fc
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-t7pgd
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-qf7h7
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-jzvq9
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-mpcfv
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-zgls6
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-pmbcq
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-ksrzr
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-hl9ns
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-m8m6t
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-clbrq
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-d2ww6
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-x2ch5
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-9fdkb
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-msdd5
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-l2xpt
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-8qfcl
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-p62j2
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-p4hlb
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-h7rmz
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-mfznt
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-bkh4l
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-master-1
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-n5cbc
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-qh4fr
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-bxprw
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-vfdln
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-8x2zx
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-hfk9f
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-fqgdc
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-jhn2h
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-44j6r
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-f6j96
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-9hpfn
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-2rdbd
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-cc76s
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-6nw5k
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-4r7fc
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-sk94f
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-vkckz
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-bjnmt
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-c7k97
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-b4krw
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-r2vvz
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-fc5xv
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-wbzrc
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-sjqxc
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-mz49q
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-d-9wzvz
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-6mr6f
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-s8f74
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-897q2
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-bqz2v
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-87447
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-4nsrk
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-mss89
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-sqbnx
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-5zstc
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-sprt9
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-mjrw9
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-n4xfp
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-46vk4
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-c-2wrq8
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-xwjqd
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-x-sqfjg
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-z-ztcsj
    Type:                  Ready-In-Zone-horsley0404gcp16x-jxm4k-worker-y-fxtcv



$ oc describe anp ama | grep horsley | wc -l
103
```

so seems like its not super bad :D to know if something is ready or not

Reply:
ok @tssurya , user may need to match this count against number of zones present  if there are any misses. but it's fine for now.

Reply:
ok @tssurya , user may need to match this count against number of zones to check if there are any misses (this may not happen at all). so it's fine for now.

Reply:
yea not ideal but for now its what we could do, I'll add a TODO with @JoeAldinger to ensure we tell users how to check if things were fine or not.